Source/Core/VideoBackends/Vulkan/StagingBuffer.cpp

```diff
@@ -180,59 +162,53 @@ bool StagingBuffer::AllocateBuffer(STAGING_BUFFER_TYPE type, VkDeviceSize size,
       0,                                     // uint32_t               queueFamilyIndexCount
       nullptr                                // const uint32_t*        pQueueFamilyIndices
   };
-  VkResult res =
-      vkCreateBuffer(g_vulkan_context->GetDevice(), &buffer_create_info, nullptr, out_buffer);
-  if (res != VK_SUCCESS)
-  {
-    LOG_VULKAN_ERROR(res, "vkCreateBuffer failed: ");
-    return false;
-  }
-
-  VkMemoryRequirements requirements;
-  vkGetBufferMemoryRequirements(g_vulkan_context->GetDevice(), *out_buffer, &requirements);
 
-  u32 type_index;
-  if (type == STAGING_BUFFER_TYPE_UPLOAD)
-    type_index = g_vulkan_context->GetUploadMemoryType(requirements.memoryTypeBits, out_coherent);
-  else
-    type_index = g_vulkan_context->GetReadbackMemoryType(requirements.memoryTypeBits, out_coherent);
-
-  VkMemoryAllocateInfo memory_allocate_info = {
-      VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO,  // VkStructureType    sType
-      nullptr,                                 // const void*        pNext
-      requirements.size,                       // VkDeviceSize       allocationSize
-      type_index                               // uint32_t           memoryTypeIndex
-  };
-  res = vkAllocateMemory(g_vulkan_context->GetDevice(), &memory_allocate_info, nullptr, out_memory);
-  if (res != VK_SUCCESS)
+  VmaAllocationCreateInfo alloc_create_info = {};
+  alloc_create_info.flags = type == STAGING_BUFFER_TYPE_UPLOAD ?
```

Comment:
`VMA_ALLOCATION_CREATE_MAPPED_BIT`?

Reply:
We could do that but it really makes no difference.
If we pass that flag, VMA will map the resource immediately for us and we can retrieve the pointer from the vmaAllocationInfo.
Right now we just do the mapping manually with `vmaMapAllocation`.
Both are perfectly valid approaches.

I can change it if you want me to but it really doesn't matter.

Reply:
I see what you mean.  Well, on the off chance VMA does something in the future with the info, it might be nice.  Not a blocker though.

Reply:
VMA already has the info that we want it to be host visible. I'll keep it as it is.